### PR TITLE
restore rollbackSnapshot

### PR DESCRIPTION
**Pull request by Steve-Mcl**
fizes #2026

## Description

restore rollbackSnapshot

## TODO
- [x] capture this in a frontend api unit test (OR raise an issue to pick this up at a later time)
   - Issue raised: https://github.com/flowforge/flowforge/issues/2032

## Related Issue(s)

#2026

## Acceptance 

 - [ ] Test operation on staging before updating prod

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass
    - Issue raised: https://github.com/flowforge/flowforge/issues/2032
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `flowforge/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `flowforge/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: frontend/src/api/instances.js [hunk 1]
```diff
@@ -127,10 +127,18 @@ const getInstanceDevices = async (instanceId, cursor, limit) => {
 const getInstanceDeviceSettings = async (instanceId) => {
     return client.get(`/api/v1/projects/${instanceId}/devices/settings`).then(res => res.data)
 }
+
 const updateInstanceDeviceSettings = async (instanceId, settings) => {
     return client.post(`/api/v1/projects/${instanceId}/devices/settings`, settings).then(res => res.data)
 }
 
+const rollbackInstance = async (projectId, snapshotId) => {
```
Comment (Pezmc) [inline frontend/src/api/instances.js:135]: /s/projectId/instanceId

**Review by Pezmc — APPROVED** [review]
Comment around renaming, but not going to block!

File: frontend/src/api/instances.js (hunk 1 above)
Comment (Steve-Mcl) [reply frontend/src/api/instances.js:135]: sorry @pez, I dont understand?Hi Pez,  this fix works locally and follows suit with all the other functions in the file.

I realise this should be (will be eventually renamed)

**Comment by Steve-Mcl** [comment]
@Pezmc could you please pick up the TODO if I raise an issue?

File: frontend/src/api/instances.js (hunk 1 above)
Comment (knolleary) [reply frontend/src/api/instances.js:135]: > follows suit with all the other functions in the file.

No it doesn't. All other functions use `instanceId`.

File: frontend/src/api/instances.js [hunk 2]
```diff
@@ -127,10 +127,18 @@ const getInstanceDevices = async (instanceId, cursor, limit) => {
...
+    const data = {
+        snapshot: snapshotId
+    }
+    return client.post(`/api/v1/projects/${projectId}/actions/rollback`, data).then(res => res.data)
+}
```
Comment (knolleary) [inline frontend/src/api/instances.js:140]:
```suggestion
const rollbackInstance = async (instanceId, snapshotId) => {
    const data = {
        snapshot: snapshotId
    }
    return client.post(`/api/v1/projects/${instanceId}/actions/rollback`, data).then(res => res.data)
}
```

**Review by knolleary — CHANGES_REQUESTED** [review]

File: frontend/src/api/instances.js (hunk 1 above)
Comment (Steve-Mcl) [reply frontend/src/api/instances.js]: sorry, I meant I followed suit with the source call

https://github.com/flowforge/flowforge/blob/2cfeb77378110f8297e6cb603c501f5d4642be6f/frontend/src/api/projectSnapshots.js#L27

and other functions

https://github.com/flowforge/flowforge/blob/2cfeb77378110f8297e6cb603c501f5d4642be6f/frontend/src/api/projectSnapshots.js#L10

Comment (Steve-Mcl) [reply frontend/src/api/instances.js]: I have also renamed this in the source function call too in commit [a8eb848](https://github.com/flowforge/flowforge/pull/2030/commits/a8eb8487bd5f8027966ef3b64a4dc2d9fe01f4a4)

**Comment by knolleary** [comment]
@Pezmc the TODO is the fact this PR has no unit test to cover the new api being added.

**Comment by Steve-Mcl** [comment]
> This needs a basic happy path E2E acceptance test to ensure rollback instance is working and we don't his this regression again, happy to help write it if needed!

- [ ] Ensure the "missing `rollbackSnapshot` function" is caught in an API test to prevent regression

**Comment by Pezmc** [comment]
@Steve-Mcl @knolleary And I spoke in a Huddle, happy for this PR to move forward now so we can backport a fix, but we must address #2032 as a follow up.

**Review by knolleary — APPROVED** [review]

**Review by Pezmc — APPROVED** [review]

**Pull request merged**